servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/ClientStrategyInfluencerChainBuilder.java

```diff
@@ -67,50 +68,70 @@ void add(HttpLoadBalancerFactory<?> lb) {
     }
 
     private void add(String purpose, ExecutionStrategyInfluencer<?> influencer, HttpExecutionStrategy strategy) {
+        assert offloadNever() != strategy && defaultStrategy() != strategy :
+                "illegal " + purpose + " required strategy (" + strategy + ") for " + influencer;
```

Comment:
If we are doing automatic replacement, not sure assertion is worth being here. Better to test that we are doing replacement correctly.

Reply:
The intention was that at least tests would fail and users would have some incentive to correct the problem. I suspect that when we eventually do disallow the incorrect strategies there will be failures due to client code not fixing this issue. We need to incrementally move towards disallowing the illegal strategies. I have removed the `assert` for now.